### DISCUSSION TITLE
1.41 - fixing stupid mistakes edition

### DISCUSSION THREAD

**Discussion by cerebrate** [arkane-systems/genie#162] [General]
* works even on non-custom kernels
  * distinguishes slow-start from no-start

<hr /><em>This discussion was created from the release <a href='https://github.com/arkane-systems/genie/releases/tag/v1.41'>1.41 - fixing stupid mistakes edition</a>.</em>

**Comment by tomdecouttere** [comment]
genie -s doesn't work in Ubuntu 21.04,
this is my output, after an hour waiting for systemd, it returns to root and the verify commands don't anything
![image](https://user-images.githubusercontent.com/45790756/117444175-648eaa00-af39-11eb-9367-6bf9320d6ca7.png)

**Reply by cerebrate** [reply]
Please make bug reports in the form of an issue. In addition to the other requested information, please run genie as `genie -v -s` for verbose output and include the output of `ps aux` and `systemctl status` after invoking it; i.e., while as root in the included screenshot.

**Comment by tomdecouttere** [comment]
![image](https://user-images.githubusercontent.com/45790756/117487775-d1bb3300-af6b-11eb-986e-12b475b30c91.png)

It seems like it is related to Ubuntu 21.04 and not Genie.
I'm not sure what Transport endpoint is not connected means.
I had understood that adding the first 2 lines would make the genie -s run
In 20.04 i think this worked.

Tnx

Op vr 7 mei 2021 om 16:30 schreef Alistair Young ***@***.***>:

**Reply by tomdecouttere** [reply]
genie -i works after the 1st run it first also waits for systemd and then a 2nd time, this is the output : 

tom@CMEC-LT-001:~$ ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0    892   576 ?        Sl   19:50   0:00 /init
root         6  0.0  0.0    892    80 ?        Ss   19:50   0:00 /init
root         7  0.0  0.0    892    80 ?        R    19:50   0:00 /init
tom          8  0.0  0.1  21724  5212 pts/0    Ss   19:50   0:00 -bash
tom         40  0.0  0.0  22948  3300 pts/0    R+   19:52   0:00 ps aux
tom@CMEC-LT-001:~$ genie -i
Waiting for systemd....^C
tom@CMEC-LT-001:~$ genie -i
tom@CMEC-LT-001:~$ ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0    892   576 ?        Sl   19:50   0:00 /init
root           6  0.0  0.0    892    80 ?        Ss   19:50   0:00 /init
root           7  0.0  0.0    892    80 ?        R    19:50   0:00 /init
tom            8  0.0  0.1  21724  5216 pts/0    Ss   19:50   0:00 -bash
root          52  0.0  0.0  18640   760 ?        Ss   19:53   0:00 /usr/bin/unshare -fp --propagation shared --mount-pro
root          53  103  0.2  34912 10864 ?        Rs   19:53   0:14 systemd
root         102 48.5  1.1  72892 48232 ?        S<s  19:53   0:06 /lib/systemd/systemd-journald
root         122  0.5  0.1  21184  5768 ?        Ss   19:53   0:00 /lib/systemd/systemd-udevd
root         140  0.0  0.0   3520   216 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/core18_1988.snap /snap/
root         141  0.0  0.0   3520   208 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/chromium_1523.snap /sna
root         142  0.0  0.0   3520   184 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/chromium_1564.snap /sna
root         144  0.0  0.0   3520   172 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/core18_1997.snap /snap/
root         146  0.0  0.0   3520   172 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/gnome-3-28-1804_145.sna
root         150  0.0  0.0   3520   172 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/gtk-common-themes_1514.
root         157  0.0  0.0   3520   220 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/gtk-common-themes_1515.
root         161  0.0  0.0   3520   184 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/hollywood_3.snap /snap/
root         166  0.0  0.0   3520   216 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/lxd_19188.snap /snap/lx
root         168  0.0  0.0   3520   228 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/lxd_19647.snap /snap/lx
root         171  0.0  0.0   3520   216 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/snapd_11402.snap /snap/
root         172  0.0  0.0   3520   168 ?        Ss   19:53   0:00 snapfuse /var/lib/snapd/snaps/snapd_11588.snap /snap/
systemd+     194  0.5  0.1  16136  6872 ?        Ss   19:53   0:00 /lib/systemd/systemd-networkd
root         195  0.0  0.1 177020  6564 ?        Ssl  19:53   0:00 /usr/lib/accountsservice/accounts-daemon
tom          214  0.0  0.0  22948  3492 pts/0    R+   19:53   0:00 ps aux

**Reply by tomdecouttere** [reply]
Within WSL : 

tom@CMEC-LT-001-wsl:~$ genie -v -s
genie: outside bottle, systemd pid: 546.
genie: starting shell
Failed to get shell PTY: Transport endpoint is not connected
genie: starting shell failed; machinectl shell returned 1.

**Comment by tomdecouttere** [comment]
Re-installed Ubuntu 20.04 LTS, the "Transport endpoint is not connected" is gone, and it works again.
Strange, but after the following screen where genie has created the shell, you have to type exit and then the screen of the Xserver starts up.
![image](https://user-images.githubusercontent.com/45790756/118833043-6ce3cf00-b8c1-11eb-9743-657e949c40e5.png)

**Comment by tomdecouttere** [comment]
>>Strange, but after the following screen where genie has created the shell, you have to type exit and then the screen of the Xserver starts up.<<
Hey, just adding "host.exit()" to the startup.sh script solved that too....
Not a Genie problem so.
